### PR TITLE
WIP Separate platform & target

### PR DESCRIPTION
**Pull request by sdmaclea**
Extend cross component build make system to allow cross OS build for some components.

Standardize on `Platform` and `Target` for the nomenclature (rename `_Host_` to `_Platform_`)
Add `CLR_CMAKE_TARGET_*`
Modify feature flag definition to depend on `CLR_CMAKE_TARGET_*`
Modify crosscomponent make to handle `CLR_CMAKE_TARGET_OS` 
Set `CLR_CMAKE_TARGET_OS` in build.cmd
Add flag to set target OS to linux.
Remove stray sos reference

### CODE REVIEW

**Comment by janvorli** [comment]
@sdmaclea it is not clear to me how we can ever build native components cross os, can you please share more details on how are we planning to do that?

**Comment by sdmaclea** [comment]
@janvorli This work is experimental at this point.

File: src/coreclr/build.cmd [hunk 1]
```diff
@@ -128,6 +128,8 @@ if /i "%1" == "-x86"                 (set __BuildArchX86=1&set processedArgs=!pr
 if /i "%1" == "-arm"                 (set __BuildArchArm=1&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
 if /i "%1" == "-arm64"               (set __BuildArchArm64=1&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
 
+if /i "%1" == "-linux"               (set __BuildOS=Linux&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
```
Comment (sdmaclea) [inline src/coreclr/build.cmd]: This option may imply general support.  Maybe a special purpose flag would be better.

**Comment by sdmaclea** [comment]
This prototype worked, but this code needs to be polished before merging.  Closing this in favor of smaller actionable PRs.

**Pull request closed without merging**